cloud-engineering-with-gcp/python/content/lab-03/README.md

```diff
@@ -0,0 +1,320 @@
+# Build and Deploy a Container on Google Cloud Run
+
+In our last lab, we deployed some static HTML files to a Google Cloud Storage bucket. In the third lab of this workshop, we're going create a Docker image to run our website on NGINX and deploy it to run on [Google Cloud Run](https://cloud.google.com/run/).
+
+## Step 1 &mdash; Create a Docker Image
+
+We'll use the [Pulumi Docker provider](https://www.pulumi.com/registry/packages/docker/) to build a Docker image we can push to Google Container Registry (GCR).
+
+Before we add anything to our project, let's make sure we configure the Docker CLI to push to GCR. We can do that by running the following command:
+
+```bash
+gcloud auth configure-docker
+```
+
+We'll also need to install the [Pulumi Command provider](https://www.pulumi.com/registry/packages/command/) so we can build our image. We can do this using `pip`. Add the following line to the bottom of your `requirements.txt`:
```

Comment:
Is there a reason you're using this provider, which isn't GA, instead of the Docker provider? While Docker might have some problems, the provider is GA and has worked just fine for this use case when working with Python in the past (the main error was here: https://github.com/pulumi/pulumi-docker/issues/336).

Reply:
Yes: `docker.Image` is unusable with GCP CloudRun on an Apple Silicon Mac. There's no way to get `docker build` to create an amd64 image and CloudRun cannot run arm64 images. See: https://github.com/pulumi/pulumi-docker/issues/310

Reply:
BTW, it would be my strong preference to use `docker.Image`. This was the only way I could get the demo code to work.